sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryPolicyWithDiagnostics.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class RetryPolicyWithDiagnostics implements IRetryPolicy{
```

Comment:
We could remove IRetryPolicy interface altogether and just use this abstract class as RetryPolicy

Reply:
Thoughts ?

Reply:
This provides default implementations for all APIs of IRetryPolicy. 
If we don't plan to use IRetryPolicy altogether, then its fine, but I would suggest keeping it in the code, as it doesn't do any harm, and opens possibilities in future for more development on this if needed.

Reply:
This provides default implementations for all APIs of IRetryPolicy.
If we don't plan to use IRetryPolicy altogether, then its fine, but I would suggest keeping it in the code, as it doesn't do any harm, and opens possibilities in future for more development on this if needed.

Reply:
Yea it doesn't do any harm , will keep it as it is